library/spdm_common_lib/libspdm_com_context_data.c

```diff
@@ -1575,32 +1611,28 @@ libspdm_return_t libspdm_append_message_f(void *context, void *session_info,
                         return LIBSPDM_STATUS_CRYPTO_ERROR;
                     }
                 } else {
-                    if (spdm_context->connection_info.peer_used_cert_chain[slot_id].buffer_hash_size
-                        != 0) {
-                        hash_size =
-                            spdm_context->connection_info.peer_used_cert_chain[slot_id].
-                            buffer_hash_size;
-                        libspdm_copy_mem(mut_cert_chain_buffer_hash,
-                                         sizeof(mut_cert_chain_buffer_hash),
-                                         spdm_context->connection_info.peer_used_cert_chain[slot_id].buffer_hash,
-                                         hash_size);
-                    } else {
-                        result = libspdm_get_peer_cert_chain_buffer(
+                    if (slot_id == 0xFF) {
+                        result = libspdm_get_peer_public_key_buffer(
                             spdm_context,
                             (const void **)&mut_cert_chain_buffer,
                             &mut_cert_chain_buffer_size);
                         if (!result) {
-                            return LIBSPDM_STATUS_INVALID_STATE_PEER;
+                            return false;
```

Comment:
why `return false`?

Reply:
I'll fix it.

Reply:
Fixed in https://github.com/DMTF/libspdm/pull/1486/commits/4e2a94599a97f52a050c15b133b1ccdb1b224336 .